### PR TITLE
feat(mysql)!: drop 5.7 support

### PR DESCRIPTION
**Pull request by lohart13**
## Pull Request Checklist

- [x] Have you added new tests to prevent regressions?
- [ ] If a documentation update is necessary, have you opened a PR to [the documentation repository](https://github.com/sequelize/website/)? 
- [x] Did you update the typescript typings accordingly (if applicable)?
- [x] Does the description below contain a link to an existing issue (Closes #[issue]) or a description of the issue you are solving?
- [x] Does the name of your PR follow [our conventions](https://github.com/sequelize/sequelize/blob/main/CONTRIBUTING.md#6-commit-your-modifications)?

## Description Of Change

This PR drops Mysql version 5.7 support and sets the mininum supported version to 8.0.19.
Version 5.7 was EOL on 31 Oct 2020 and extended support ended on 31 Oct 2023.

## Todos

- [ ] Update version support table on sequelize website
- [ ] Add any other new features that MySql v8 now supports

### CODE REVIEW

File: packages/core/src/dialects/mariadb/index.ts [hunk 1]
```diff
@@ -107,7 +107,7 @@ export class MariaDbDialect extends AbstractDialect {
   }
 
   escapeString(value: string) {
-    return escapeMysqlString(value);
+    return escapeMariaDbString(value);
```
Comment (ephys) [inline packages/core/src/dialects/mariadb/index.ts:110]: considering it's the same function, I wouldn't duplicate the code. Maybe rename the function to `escapeMysqlMariaDbString` instead?

**Review by ephys — CHANGES_REQUESTED** [review]
Only note

Thank you for this :)

File: packages/core/src/dialects/mariadb/index.ts (hunk 1 above)
Comment (lohart13) [reply packages/core/src/dialects/mariadb/index.ts]: Done

**Review by ephys — APPROVED** [review]
Thanks!

**Review by WikiRik — APPROVED** [review]

**Pull request merged**